pkg/e2e/ps_test.go

```diff
@@ -57,37 +59,34 @@ func TestPs(t *testing.T) {
 
 	t.Run("json", func(t *testing.T) {
 		res = c.RunDockerComposeCmd("-f", "./fixtures/ps-test/compose.yaml", "--project-name", projectName, "ps", "--format", "json")
-		var output []map[string]interface{}
+		var output []api.ContainerSummary
 		err := json.Unmarshal([]byte(res.Combined()), &output)
 		assert.NoError(t, err)
 
 		count := 0
 		assert.Equal(t, 2, len(output))
 		for _, service := range output {
-			publishers := service["Publishers"].([]interface{})
-			if service["Name"] == "e2e-ps-busybox-1" {
+			publishers := service.Publishers
+			if service.Name == "e2e-ps-busybox-1" {
```

Comment:
Are the services expected to be returned in a consistent order? If so, this loop may not be needed, and possibly it could be a single compare between an expected slice (in the right order)?

Reply:
Ah, I guess there may be other fields that need to be ignored in the comparing in that case 🤔

Reply:
Yeah, the order may be consistent, I'm not familiar enough with the way the summaries are generated to know for sure. I was a little surprised that they were reported in alpha order instead of declaration order in the compose file.

Reply:
If there are any documented assertions about order then I'll update this to match.

Reply:
It was mostly a "drive-by" comment from my side; perhaps others more familiar with the codebase could provide more details 😅 👍